src/vs/workbench/services/hover/browser/hoverWidget.ts

```diff
@@ -443,6 +434,22 @@ export class HoverWidget extends Widget {
 		}
 	}
 
+	private adjustHoverMaxHeight(target: TargetRect): void {
+		let maxHeight = 250;
```

Comment:
There's probably a better starting value than this, especially for larger screens.